### ISSUE TITLE
Why are all IPs hardcoded to 172.19.199.* ?

### ISSUE THREAD

**Issue by samtuke** [maxking/docker-mailman#366]
Am I missing something here? All users have had to manually go and change over a dozen references to IP 172.19.199.*? Surely these should be set centrally in single environment variable?

**Comment by maxking** [comment]
Almost all the addresses that are hardcoded can be changed via Environment addresses. The reason they are hardcoded is to provide a reasonbale default that works with the provided docker-compose.yml file.

If it coincides with some network you already have, please update the compose file and set the right environment variables, all of them should be defined in the docs.

**Comment by pini-gh** [comment]
There is no need at all for these hardcoded IPs provided that the containers are given hostnames to identify each other. Here a functional docker-compose.yml file (variables defined into a separate `.env` file):
```
version: '2'

services:
  mailman-core:
    image: maxking/mailman-core:0.3
    container_name: mailman-core
    hostname: mailman-core
    restart: always
    volumes:
    - core-vol:/opt/mailman
    stop_grace_period: 30s
    links:
    - database:database
    depends_on:
    - database
    environment:
    - DATABASE_URL
    - DATABASE_TYPE
    - DATABASE_CLASS
    - HYPERKITTY_API_KEY
    - SMTP_HOST
    - SMTP_PORT
    # To prevent entrypoint setting this with 'hostname -i' which may return multiple IPs
    - MM_HOSTNAME=mailman-core
    networks:
      bridge:

  mailman-web:
    image: maxking/mailman-web:0.3
    container_name: mailman-web
    hostname: mailman-web
    restart: always
    depends_on:
    - database
    links:
    - mailman-core:mailman-core
    - database:database
    volumes:
    - web-vol:/opt/mailman-web-data
    environment:
    - DATABASE_TYPE
    - DATABASE_URL
    - HYPERKITTY_API_KEY
    - SMTP_HOST
    - SMTP_PORT
    - SERVE_FROM_DOMAIN
    - MAILMAN_ADMIN_USER
    - MAILMAN_ADMIN_EMAIL
    - SECRET_KEY
    - UWSGI_STATIC_MAP=/static=/opt/mailman-web-data/static
    networks:
      bridge:

 database:
    container_name: mailman-database
    environment:
      - POSTGRES_DB
      - POSTGRES_USER
      - POSTGRES_PASSWORD
    restart: always
    image: postgres:9.6-alpine
    volumes:
    - database-vol:/var/lib/postgresql/data
    networks:
      bridge:

volumes:
  core-vol:
  web-vol:
  database-vol:

networks:
   bridge: bridge
```

**Comment by danil-smirnov** [comment]
What do you mean by 'functional docker-compose.yml file' @pini-gh ? Did you manage to up the whole thing without a single IP addresses in the configuration?

I believe it won't work as `MAILMAN_HOST_IP` variable which you have missed must be set to IP address of Core container to whitelist it in Hyperkitty...

And the latter won't work until [this issue](https://gitlab.com/mailman/hyperkitty/-/issues/157) resolved.

**Comment by pini-gh** [comment]
> What do you mean by 'functional docker-compose.yml file' @pini-gh ? Did you manage to up the whole thing without a single IP addresses in the configuration?
Yes I did. It is up and running.

> I believe it won't work as `MAILMAN_HOST_IP` variable which you have missed must be set to IP address of Core container to whitelist it in Hyperkitty...
Please elaborate. It's not impossible there are no shortcomings in my configuration, but I don't get what you mean here.




> What do you mean by 'functional docker-compose.yml file' @pini-gh ? Did you manage to up the whole thing without a single IP addresses in the configuration?
> 
> I believe it won't work as `MAILMAN_HOST_IP` variable which you have missed must be set to IP address of Core container to whitelist it in Hyperkitty...

Oh, you're right indeed:
```
ERROR 2020-04-13 19:01:02,458 20 hyperkitty.views.mailman Access to the archiving API endpoint was forbidden from IP 172.27.0.5, your MAILMAN_ARCHIVER_FROM setting may be misconfigured
```
These logs don't show up in docker logs.

I don't see the point in filterring by IPs while there is already the HYPERKITTY_API_KEY.

Anyway, I've just workaround-ed that using a custom `settings_local.py` with:
```
MAILMAN_ARCHIVER_FROM = (gethostbyname("mailman-core"),)
```

**Comment by epifanio** [comment]
HI!  - not an expert here, sorry if it is a dumb question - I am trying to run the docker-mailman setup as part of my docker swarm.
 
If I well-understood docker-swarm networking, it seems doesn't allow static-IP assignment to the containers.

May I post my issue and related dockerfile here - or better I open a new issue?

 I can access the mailman-web interface which is proxied correctly with https via traefik - but when attempt to reset the password the game ends with an internal error. - looking at the services logs, seems my error (at the moment) is with mailman-core:

```
root@epinux /home/epinux/dev/docker_swarm # docker service logs lists_mailman-core
lists_mailman-core.1.8j2hry60cx18@epinux.com    |                                List of databases
lists_mailman-core.1.8j2hry60cx18@epinux.com    |    Name    |  Owner  | Encoding |  Collate   |   Ctype    |  Access privileges  
lists_mailman-core.1.8j2hry60cx18@epinux.com    | -----------+---------+----------+------------+------------+---------------------
lists_mailman-core.1.8j2hry60cx18@epinux.com    |  mailmandb | mailman | UTF8     | en_US.utf8 | en_US.utf8 | 
lists_mailman-core.1.8j2hry60cx18@epinux.com    |  postgres  | mailman | UTF8     | en_US.utf8 | en_US.utf8 | 
lists_mailman-core.1.8j2hry60cx18@epinux.com    |  template0 | mailman | UTF8     | en_US.utf8 | en_US.utf8 | =c/mailman         +
lists_mailman-core.1.8j2hry60cx18@epinux.com    |            |         |          |            |            | mailman=CTc/mailman
lists_mailman-core.1.8j2hry60cx18@epinux.com    |  template1 | mailman | UTF8     | en_US.utf8 | en_US.utf8 | =c/mailman         +
lists_mailman-core.1.8j2hry60cx18@epinux.com    |            |         |          |            |            | mailman=CTc/mailman
lists_mailman-core.1.8j2hry60cx18@epinux.com    | (4 rows)
lists_mailman-core.1.8j2hry60cx18@epinux.com    | 
lists_mailman-core.1.8j2hry60cx18@epinux.com    | Postgres is up - continuing
lists_mailman-core.1.8j2hry60cx18@epinux.com    | No MTA environment variable found, defaulting to Exim
lists_mailman-core.1.8j2hry60cx18@epinux.com    | Found configuration file at /opt/mailman/mailman-extra.cfg
lists_mailman-core.1.8j2hry60cx18@epinux.com    | Traceback (most recent call last):
lists_mailman-core.1.8j2hry60cx18@epinux.com    |   File "/usr/bin/runner", line 8, in <module>
lists_mailman-core.1.8j2hry60cx18@epinux.com    |     sys.exit(main())
lists_mailman-core.1.8j2hry60cx18@epinux.com    |   File "/usr/lib/python3.8/site-packages/click/core.py", line 829, in __call__
lists_mailman-core.1.8j2hry60cx18@epinux.com    |     return self.main(*args, **kwargs)
lists_mailman-core.1.8j2hry60cx18@epinux.com    |   File "/usr/lib/python3.8/site-packages/click/core.py", line 782, in main
lists_mailman-core.1.8j2hry60cx18@epinux.com    |     rv = self.invoke(ctx)
lists_mailman-core.1.8j2hry60cx18@epinux.com    |   File "/usr/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
lists_mailman-core.1.8j2hry60cx18@epinux.com    |     return ctx.invoke(self.callback, **ctx.params)
lists_mailman-core.1.8j2hry60cx18@epinux.com    |   File "/usr/lib/python3.8/site-packages/click/core.py", line 610, in invoke
lists_mailman-core.1.8j2hry60cx18@epinux.com    |     return callback(*args, **kwargs)
lists_mailman-core.1.8j2hry60cx18@epinux.com    |   File "/usr/lib/python3.8/site-packages/click/decorators.py", line 21, in new_func
lists_mailman-core.1.8j2hry60cx18@epinux.com    |     return f(get_current_context(), *args, **kwargs)
lists_mailman-core.1.8j2hry60cx18@epinux.com    |   File "/usr/lib/python3.8/site-packages/mailman/bin/runner.py", line 188, in main
lists_mailman-core.1.8j2hry60cx18@epinux.com    |     runner.run()
lists_mailman-core.1.8j2hry60cx18@epinux.com    |   File "/usr/lib/python3.8/site-packages/mailman/runners/lmtp.py", line 249, in run
lists_mailman-core.1.8j2hry60cx18@epinux.com    |     self.lmtp.start()
lists_mailman-core.1.8j2hry60cx18@epinux.com    |   File "/usr/lib/python3.8/site-packages/aiosmtpd/controller.py", line 60, in start
lists_mailman-core.1.8j2hry60cx18@epinux.com    |     raise self._thread_exception
lists_mailman-core.1.8j2hry60cx18@epinux.com    |   File "/usr/lib/python3.8/site-packages/aiosmtpd/controller.py", line 37, in _run
lists_mailman-core.1.8j2hry60cx18@epinux.com    |     self.server = self.loop.run_until_complete(
lists_mailman-core.1.8j2hry60cx18@epinux.com    |   File "/usr/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
lists_mailman-core.1.8j2hry60cx18@epinux.com    |     return future.result()
lists_mailman-core.1.8j2hry60cx18@epinux.com    |   File "/usr/lib/python3.8/asyncio/base_events.py", line 1463, in create_server
lists_mailman-core.1.8j2hry60cx18@epinux.com    |     raise OSError(err.errno, 'error while attempting '
lists_mailman-core.1.8j2hry60cx18@epinux.com    | OSError: [Errno 99] error while attempting to bind on address ('172.19.199.2', 8024): address not available
```

**Comment by pini-gh** [comment]
Mailman-web logs are into `/opt/mailman-web-data/logs/`.

What does the error look like? Can you post your - redacted - `docker-compose.yml`?

**Comment by epifanio** [comment]
Hi!

my docker-compose for the swarm looks like this:

```yaml
version: '3.7'

services:
  mailman-core:
    image: maxking/mailman-core:0.3
    # container_name: mailman-core
    hostname: mailman-core
    volumes:
      - /opt/mailman/core:/opt/mailman/
    stop_grace_period: 30s
    links:
      - database:database
    depends_on:
      - database
    environment:
      - MM_HOSTNAME=mailman-core
      - DATABASE_URL=postgres://mailman:mailmanpass@database/mailmandb
      - DATABASE_TYPE=postgres
      - DATABASE_CLASS=mailman.database.postgresql.PostgreSQLDatabase
      - HYPERKITTY_URL=lists.mydomain.com 
      - HYPERKITTY_API_KEY=----------------------------------------------------------------
    networks:
      mailman:
        ipv4_address: 172.19.199.2

  mailman-web:
    image: maxking/mailman-web:0.3
    # container_name: mailman-web
    hostname: mailman-web
    ports:
      - 12435:8000
    deploy:
      resources:
        #limits:
        #  cpus: '0.50'
        #  memory: 2048M
        reservations:
          cpus: '0.25'
          memory: 512M
      placement:
        constraints:
          - node.role == manager
      labels:
        - traefik.enable=true
        - traefik.docker.network=traefik-public
        - traefik.constraint-label=traefik-public
        - traefik.http.routers.lists-http.rule=Host(`${DOMAIN?Variable not set}`)
        - traefik.http.routers.lists-http.entrypoints=http
        - traefik.http.routers.lists-http.middlewares=https-redirect
        - traefik.http.routers.lists-https.rule=Host(`${DOMAIN?Variable not set}`)
        - traefik.http.routers.lists-https.entrypoints=https
        - traefik.http.routers.lists-https.tls=true
        - traefik.http.routers.lists-https.tls.certresolver=le
        - traefik.http.services.lists.loadbalancer.server.port=8000
        # allow CORS
        - "traefik.http.middlewares.testheader.headers.accesscontrolallowmethods=GET,OPTIONS,PUT"
        - "traefik.http.middlewares.testheader.headers.accesscontrolalloworiginlist=*"
        - "traefik.http.middlewares.testheader.headers.accesscontrolmaxage=100"
        - "traefik.http.middlewares.testheader.headers.addvaryheader=true"
    depends_on:
      - database
    links:
      - mailman-core:mailman-core
      - database:database
    volumes:
      - /opt/mailman/web:/opt/mailman-web-data
      # - ./settings_local.py:/opt/mailman-web/settings_local.py
    environment:
      - DATABASE_TYPE=postgres
      - DATABASE_URL=postgres://mailman:mailmanpass@database/mailmandb
      - HYPERKITTY_API_KEY=-----------------------------------------------------------------------------------
      - UWSGI_STATIC_MAP=/static=/opt/mailman-web-data/static
      - SERVE_FROM_DOMAIN=lists.maydomain.com
      - MAILMAN_ADMIN_USER=admin
      - MAILMAN_ADMIN_EMAIL=email@gmail.com
      - SECRET_KEY=password
    networks:
      mailman:
        ipv4_address: 172.19.199.3
      traefik-public:

  database:
    environment:
      POSTGRES_DB: mailmandb
      POSTGRES_USER: mailman
      POSTGRES_PASSWORD: mailmanpass
    image: postgres:9.6-alpine
    volumes:
      - /opt/mailman/database:/var/lib/postgresql/data
    networks:
      mailman:
        ipv4_address: 172.19.199.4

networks:
  net:
    driver: overlay
    attachable: true
  traefik-public:
    external: true
  mailman:
    attachable: true
    driver: overlay
    ipam:
      driver: default
      config:
      -
        subnet: 172.19.199.0/24
```

I exposed the web app on port: 12435
I changed the mailman network driver to: overlay to overcome the error below when running the service:
```
failed to create service lists_mailman-core: Error response from daemon: The network lists_mailman cannot be used with services. Only networks scoped to the swarm can be used, such as those created with the overlay driver.
```
on the host machine I changed the following files:

* `/opt/mailman/web/settings_local.py`
```
EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
EMAIL_HOST = '172.19.199.1'
EMAIL_PORT = 25
USE_SSL = False
DEFAULT_FROM_EMAIL = "mailman@lists.mydomain.com"
SERVER_EMAIL = "mailman@lists.mydomain.com"
```

* `/opt/mailman/core/mailman-extra.cfg`

```
[mta]
incoming: mailman.mta.postfix.LMTP
outgoing: mailman.mta.deliver.deliver
lmtp_host: 172.19.199.3
lmtp_port: 8024
smtp_host: 172.19.199.1
smtp_port: 25
configuration: /etc/postfix-mailman.cfg
```
* `/etc/postfix/main.cf`

```
# See /usr/share/postfix/main.cf.dist for a commented, more complete version


# Debian specific:  Specifying a file name will cause the first
# line of that file to be used as the name.  The Debian default
# is /etc/mailname.
#myorigin = /etc/mailname

smtpd_banner = $myhostname ESMTP $mail_name (Ubuntu)
biff = no

# appending .domain is the MUA's job.
append_dot_mydomain = no

# Uncomment the next line to generate "delayed mail" warnings
#delay_warning_time = 4h

readme_directory = no

# See http://www.postfix.org/COMPATIBILITY_README.html -- default to 2 on
# fresh installs.
compatibility_level = 2



# TLS parameters
smtpd_tls_cert_file=/etc/ssl/certs/ssl-cert-snakeoil.pem
smtpd_tls_key_file=/etc/ssl/private/ssl-cert-snakeoil.key
smtpd_tls_security_level=may

smtp_tls_CApath=/etc/ssl/certs
smtp_tls_security_level=may
smtp_tls_session_cache_database = btree:${data_directory}/smtp_scache


smtpd_relay_restrictions = permit_mynetworks permit_sasl_authenticated defer_unauth_destination
myhostname = Ubuntu-1910-eoan-64-minimal
alias_maps = hash:/etc/aliases
alias_database = hash:/etc/aliases
mydestination = $myhostname, Ubuntu-1910-eoan-64-minimal, localhost.localdomain, , localhost
relayhost = 
# mynetworks = 127.0.0.0/8 [::ffff:127.0.0.0]/104 [::1]/128
mailbox_size_limit = 0
recipient_delimiter = +
inet_interfaces = all
inet_protocols = all

# main.cf
# mynetworks erweitern um 172.19.199.2 und 172.19.199.3
# Support the default VERP delimiter.
# recipient_delimiter = +
mynetworks = 172.19.199.0/8 [::ffff:172.19.199.0]/104 [::1]/128
unknown_local_recipient_reject_code = 550
owner_request_special = no
 
transport_maps =
    regexp:/opt/mailman/core/var/data/postfix_lmtp
local_recipient_maps =
    regexp:/opt/mailman/core/var/data/postfix_lmtp
relay_domains =
    regexp:/opt/mailman/core/var/data/postfix_domains
```

I checked for the logs into `/opt/mailman-web-data/` from inside the container (they are not exposed as volume) and I can see:

* ` /opt/mailman-web-data/logs/mailmanweb.log`

```
ERROR 2020-11-06 08:54:37,801 1707 django.request Internal Server Error: /accounts/password/reset/
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/usr/lib/python3.8/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/lib/python3.8/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/lib/python3.8/site-packages/django/views/generic/base.py", line 71, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/lib/python3.8/site-packages/django/views/generic/base.py", line 97, in dispatch
    return handler(request, *args, **kwargs)
  File "/usr/lib/python3.8/site-packages/allauth/account/views.py", line 104, in post
    response = self.form_valid(form)
  File "/usr/lib/python3.8/site-packages/allauth/account/views.py", line 645, in form_valid
    form.save(self.request)
  File "/usr/lib/python3.8/site-packages/allauth/account/forms.py", line 537, in save
    get_adapter(request).send_mail(
  File "/usr/lib/python3.8/site-packages/allauth/account/adapter.py", line 137, in send_mail
    msg.send()
  File "/usr/lib/python3.8/site-packages/django/core/mail/message.py", line 306, in send
    return self.get_connection(fail_silently).send_messages([self])
  File "/usr/lib/python3.8/site-packages/django/core/mail/backends/smtp.py", line 103, in send_messages
    new_conn_created = self.open()
  File "/usr/lib/python3.8/site-packages/django/core/mail/backends/smtp.py", line 63, in open
    self.connection = self.connection_class(self.host, self.port, **connection_params)
  File "/usr/lib/python3.8/smtplib.py", line 253, in __init__
    (code, msg) = self.connect(host, port)
  File "/usr/lib/python3.8/smtplib.py", line 339, in connect
    self.sock = self._get_socket(host, port, self.timeout)
  File "/usr/lib/python3.8/smtplib.py", line 308, in _get_socket
    return socket.create_connection((host, port), timeout,
  File "/usr/lib/python3.8/socket.py", line 808, in create_connection
    raise err
  File "/usr/lib/python3.8/socket.py", line 796, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused
```

* `/opt/mailman-web-data/logs/uwsgi-error.log` is quite long (most repetition of the same error), attaching it as a gist:  https://gist.github.com/epifanio/c524f978d30464ac08dd21aa00539831

**Comment by epifanio** [comment]
the following info (which shows the IP do not correspond with the ipv4 settings in the docker-compose ) could be useful:

*  `docker network inspect lists_mailman`
```
[
    {
        "Name": "lists_mailman",
        "Id": "xylf9jv338euhkbwlwehscdpc",
        "Created": "2020-11-06T10:08:25.101447929+01:00",
        "Scope": "swarm",
        "Driver": "overlay",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "172.19.199.0/24",
                    "Gateway": "172.19.199.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": true,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "484db7eea9cf97378f30b2e5bb0bf69ba460e28696704c0406f41ac79c1902c3": {
                "Name": "lists_database.1.gumumktadik295m9rrl96azbd",
                "EndpointID": "4fea2d4e665d784506226b6d61a43cde13aec450acc1af70101279e00718d790",
                "MacAddress": "02:42:ac:13:c7:08",
                "IPv4Address": "172.19.199.8/24",
                "IPv6Address": ""
            },
            "c21a8a977ed9f3e328581690c7d8139da7ddbfa47912f51af8c094bf80e168e2": {
                "Name": "lists_mailman-web.1.u1n8srn6gdsyosnhscbfsaodh",
                "EndpointID": "c8e5f7e80f96b54ef5d6af8990be1f9a80823eeca64c0028e4b860d61a5c7b90",
                "MacAddress": "02:42:ac:13:c7:06",
                "IPv4Address": "172.19.199.6/24",
                "IPv6Address": ""
            },
            "c2b0458b4a562087a1caa97e448dd8b9dcce9b2f03676ba304e73e2e7c9484b9": {
                "Name": "lists_mailman-core.1.seevixfh0ezjklsx3cxwsqs98",
                "EndpointID": "bf6abe5629f8171a31e213dcc12dd176a19168f468e8b2fde3cc7b807b5117c7",
                "MacAddress": "02:42:ac:13:c7:03",
                "IPv4Address": "172.19.199.3/24",
                "IPv6Address": ""
            },
            "lb-lists_mailman": {
                "Name": "lists_mailman-endpoint",
                "EndpointID": "53e0135ebdeed5e4dced60903a4c72e959867554ccb342c45f4f9210e3c6950a",
                "MacAddress": "02:42:ac:13:c7:04",
                "IPv4Address": "172.19.199.4/24",
                "IPv6Address": ""
            }
        },
        "Options": {
            "com.docker.network.driver.overlay.vxlanid_list": "4168"
        },
        "Labels": {
            "com.docker.stack.namespace": "lists"
        },
        "Peers": [
            {
                "Name": "81d98c4d2c5c",
                "IP": "178.63.52.22"
            }
        ]
    }
]
```

**Comment by pini-gh** [comment]
@maxking 
I've helped @epifanio working around the issue for his swarm. There is a problem with the way Hyperkitty checks for allowed archiver clients, using `MAILMAN_ARCHIVER_FROM` IP list tuple. This can't work in swarm mode, Because AFAIUI mailman-core and mailman-web IPs are not the same depending on which end you look from. For example:

```
bash-5.0# hostname
mailman-web
bash-5.0# ip a
...
569: eth1@if570: <BROADCAST,MULTICAST,UP,LOWER_UP,M-DOWN> mtu 1450 qdisc noqueue state UP 
    link/ether 02:42:0a:00:09:16 brd ff:ff:ff:ff:ff:ff
    inet 10.0.9.22/24 brd 10.0.9.255 scope global eth1
       valid_lft forever preferred_lft forever
...
bash-5.0# cat /etc/hosts | grep mailman-web
10.0.9.22	mailman-web
10.0.1.61	mailman-web
bash-5.0# nslookup mailman-web | grep Address
Address:	127.0.0.11:53
Address: 10.0.9.16
bash-5.0# nslookup mailman-core | grep Address
Address:	127.0.0.11:53
Address: 10.0.9.20
```
And :
```
bash-5.0# hostname
mailman-core
bash-5.0# ip a
...
555: eth0@if556: <BROADCAST,MULTICAST,UP,LOWER_UP,M-DOWN> mtu 1450 qdisc noqueue state UP 
    link/ether 02:42:0a:00:09:15 brd ff:ff:ff:ff:ff:ff
    inet 10.0.9.21/24 brd 10.0.9.255 scope global eth0
       valid_lft forever preferred_lft forever
...
bash-5.0# cat /etc/hosts | grep mailman-core
10.0.9.21	mailman-core
bash-5.0# nslookup mailman-core | grep Address
Address:	127.0.0.11:53
Address: 10.0.9.20
bash-5.0# nslookup mailman-web | grep Address
Address:	127.0.0.11:53
Address: 10.0.9.16
```
Then it seems impossible to define a correct value for `MAILMAN_HOST_IP` which is used into mailman-web's `settings.py` to configure `MAILMAN_ARCHIVER_FROM`.

How about an option to disable this IP check against `MAILMAN_ARCHIVER_FROM` and rely on `HYPERKITTY_API_KEY` only?

**Comment by pini-gh** [comment]
This will be fixed in next mailman release (1.3.4) which accepts the wildcard`'*'` for _any host_ in `MAILMAN_ARCHIVER_FROM`.

**Comment by codeceptsDE** [comment]
> This will be fixed in next mailman release (1.3.4) which accepts the wildcard`'*'` for _any host_ in `MAILMAN_ARCHIVER_FROM`.

Can confirm that this works now, since Hyperkitty [has been updated to 1.3.4](https://github.com/maxking/docker-mailman/commit/297bf35b1b149baba61fc975e7d59bcef82fd25d#diff-58e45409f99345d0feb8b51f3aa5e664b508a7425305f1f1e720bf372f32f435R22).
You can now add `MAILMAN_HOST_IP=*` as a mailman-web environment variable to have Hyperkitty accept any host. I understand this is still secure as it is still protected by the api key. Would be nice if this could be reflected in the README.

My setup now uses docker hostnames entirely, so I'd have to agree with @pini-gh that the images should not need to use static IPs anywhere. Maybe something like an updated #414 can get merged some day.

**Comment by maxking** [comment]
Closing this since 0.4 is now released with the images not using Static IPs. We are still using a static subnet in the provided compose files, since we need some way to allow containers to send emails via Postfix which would need allow containers to send emails and it can't keep changing depending on which subnet is picked up by docker.

Users who can configure postfix/other mta to allow sending emails from all the containers in other ways don't need this static subnet setup and can choose to go completely dynamic.
